lib/cylc/gui/gscan.py

```diff
@@ -512,9 +513,17 @@ def __init__(self, hosts=None, owner=None, poll_interval=None):
         scrolled_window.add(self.suite_treeview)
         scrolled_window.show()
         self.vbox.pack_start(scrolled_window, expand=True, fill=True)
+
+        if name:
+            name_pattern = name
+        else:
+            name_pattern = ['.*']
+        name_pattern = "(" + ")|(".join(name_pattern) + ")"
```

Comment:
You may want to compile the pattern here with `re.compile` - to catch any `re.error` that it may generate. (In addition, using a compiled regular expression object's `.match` method for matching a string should in theory be faster than using `re.match`.)